MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,430 @@
+}
+
+- (BOOL)didSuccessfullySaveContext
+{
+    NSError *error = nil;
+    if (![self.managedObjectContext save:&error]) {
+        NSLog(@"Unable to save managed object context.");
+        NSLog(@"%@, %@", error, error.localizedDescription);
+        return NO;
+    }
+    return YES;
+}
+
+- (AreaDrop *)getUniqueAreaDropWithDecimalID:(NSString *)decimalID
+{
+    NSError *error = nil;
+    NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:@"AreaDrop"];
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %@", @"idDecimalString", decimalID];
+    [request setPredicate:predicate];
+    
+    NSArray *results = [self.managedObjectContext executeFetchRequest:request error:&error];
+    if (!error) {
+        if (results.count == 0) {
+            // Object not found.
+            return nil;
+        } else if (results.count == 1) {
+            return results.firstObject;
+        } else {
+            NSLog(@"Too many entities found!");
+            return nil;
+        }
+    } else {
+        NSLog(@"Error fetching object!");
+        NSLog(@"%@, %@", error, error.localizedDescription);
+        return nil;
+    }
+
+}
+
+- (Area *)getUniqueAreaWithCombinedName:(NSString *)combinedName
+{
+    NSError *error = nil;
+    NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:@"Area"];
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %@", @"combinedName", combinedName];
+    [request setPredicate:predicate];
+    
+    NSArray *results = [self.managedObjectContext executeFetchRequest:request error:&error];
+    if (!error) {
+        if (results.count == 0) {
+            // Object not found.
+            return nil;
+        } else if (results.count == 1) {
+            return results.firstObject;
+        } else {
+            NSLog(@"Too many entities found!");
+            return nil;
+        }
+    } else {
+        NSLog(@"Error fetching object!");
+        NSLog(@"%@, %@", error, error.localizedDescription);
+        return nil;
+    }
+}
+
+- (NSManagedObject *)getUniqueEntityWithEntityName:(NSString *)entityName withNameAttribute:(NSString *)name
+{
+    NSError *error = nil;
+    NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:entityName];
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %@", @"name", name];
+    [request setPredicate:predicate];
+    
+    NSArray *results = [self.managedObjectContext executeFetchRequest:request error:&error];
+    if (!error) {
+        if (results.count == 0) {
+            // Object not found.
+            return nil;
+        } else if (results.count == 1) {
+            return results.firstObject;
+        } else {
+            NSLog(@"Too many entities found!");
+            return nil;
+        }
+    } else {
+        NSLog(@"Error fetching object!");
+        NSLog(@"%@, %@", error, error.localizedDescription);
+        return nil;
+    }
```

Comment:
Given the level of duplication of code across these `getUnique...` methods, it would probably be better to have a more-generic common method that all of these call into to handle the common aspects.